mipengine/controller/controller.py

```diff
@@ -39,13 +39,13 @@ async def exec_algorithm(
         )
 
         # TODO: AlgorithmExecutor is not yet implemented with asyncio. This is a
-        # temprorary solution for not blocking the calling function
+        # temporary solution for not blocking the calling function
         def run_algorithm_executor_in_threadpool(
             algorithm_execution_dto: AlgorithmExecutionDTO,
             all_nodes_tasks_handlers: NodesTasksHandlersDTO,
         ):
             algorithm_executor = AlgorithmExecutor(
-                algorithm_execution_dto, all_nodes_tasks_handlers
+                request_id, algorithm_execution_dto, all_nodes_tasks_handlers
```

Comment:
Since there is already an `algorithm_execution_dto` used for passing parameters into the `AlgorithmExecutor` the `request_id` should probably go in there (`AlgorithmExecutionDTO`).